frontend/javascripts/admin/auth/registration_form_wkorg.tsx

```diff
@@ -155,32 +159,57 @@ function RegistrationFormWKOrg(props: Props) {
           placeholder="Password"
         />
       </FormItem>
+      <div className="registration-form-checkboxes">
+        <FormItem
+          name="privacy_check"
+          valuePropName="checked"
+          rules={[
+            {
+              validator: (_, value) =>
+                value
+                  ? Promise.resolve()
+                  : Promise.reject(new Error(messages["auth.privacy_check_required"])),
+            },
+          ]}
+        >
+          <Checkbox>
+            I agree to storage and processing of my personal data as described in the{" "}
+            <a target="_blank" href="/privacy" rel="noopener noreferrer">
+              privacy statement
+            </a>
+            .
+          </Checkbox>
+        </FormItem>
 
-      <FormItem
-        name="privacy_check"
-        valuePropName="checked"
-        rules={[
-          {
-            validator: (_, value) =>
-              value
-                ? Promise.resolve()
-                : Promise.reject(new Error(messages["auth.privacy_check_required"])),
-          },
-        ]}
-      >
-        <Checkbox>
-          I agree to storage and processing of my personal data as described in the{" "}
-          <a target="_blank" href="/privacy" rel="noopener noreferrer">
-            privacy statement
-          </a>
-          .
-        </Checkbox>
-      </FormItem>
-      <FormItem
-        style={{
-          marginBottom: 10,
-        }}
-      >
+        {terms == null || terms.enabled ? (
+          <FormItem
+            name="tos_check"
+            valuePropName="checked"
+            rules={[
+              {
+                validator: (_, value) =>
+                  value
+                    ? Promise.resolve()
+                    : Promise.reject(new Error(messages["auth.tos_check_required"])),
+              },
+            ]}
+          >
+            <Checkbox disabled={terms == null}>
+              I agree to the{" "}
+              {terms == null ? (
+                "terms of service"
+              ) : (
+                <a target="_blank" href={terms.url} rel="noopener noreferrer">
+                  terms of service
+                </a>
+              )}
+              .
+            </Checkbox>
+          </FormItem>
+        ) : null}
+      </div>
+
+      <FormItem>
```

Comment:
Hmmm :thinking: 

This looks like quite some code duplication. Could you please reduce it in this PR before merging? 
I'd say a mini wrapping component looks ideal here:
```javascript
function TOSFromItem({terms}: props){
{terms == null || terms.enabled ? (
          <FormItem
            name="tos_check"
            valuePropName="checked"
            rules={[
              {
                validator: (_, value) =>
                  value
                    ? Promise.resolve()
                    : Promise.reject(new Error(messages["auth.tos_check_required"])),
              },
            ]}
          >
            <Checkbox disabled={terms == null}>
              I agree to the{" "}
              {terms == null ? (
                "terms of service"
              ) : (
                <a target="_blank" href={terms.url} rel="noopener noreferrer">
                  terms of service
                </a>
              )}
              .
            </Checkbox>
          </FormItem>
        ) : null}
```

What do you think?

Reply:
I totally agree. I just felt like both registration forms needed a refactoring because there is quite some code duplication, so I didnt even bother to make it DRY. but you are right, I shouldnt add more duplication.

Reply:
I extracted it :) will resolve this conversation for now, let me know if you think that anything is missing.